client/components/category-navigation/index.tsx

```diff
@@ -0,0 +1,121 @@
+import { Button } from '@wordpress/components';
+import { useEffect, useState, useRef } from '@wordpress/element';
+import classnames from 'classnames';
+
+import './style.scss';
+
+type CategoryNavProps = {
+	buttons?: {
+		icon: string;
+		label: string;
+		link: string;
+	}[];
```

Comment:
It's my bad for not realizing this is a generalized component. However, it looks like `Discover` and `All categories` will also be reused (p1709817801121679/1709817468.279259-slack-C06ELHR6L9J) in the other places where we plan to use this component. Maybe we shouldn't pass those as props after all

Reply:
`All categories` is meant to be a dropdown and not a link in and of itself, which might also support hard coding it instead of passing it as a prop BnTfb6hzKcRTb2LBtuPwKe-fi-246:42348#727280144

Reply:
>All categories is meant to be a dropdown and not a link in and of itself, which might also support hard coding it instead of passing it as a prop [BnTfb6hzKcRTb2LBtuPwKe-fi-246:42348#727280144](https://href.li/?https://www.figma.com/file/BnTfb6hzKcRTb2LBtuPwKe/?node-id=246:42348#727280144)

Hm, maybe I agree with you. A lot of questions... 😅 I propose to keep it also as is, to unblock Stephane and have some basic markup, and I will adjust it in the followup PR. I think that this way will speed up other features/functionality.

Reply:
Sure! We can definitely return to this in a later PR